app/src/main/java/org/mozilla/focus/home/HomeFragment.kt

```diff
@@ -212,6 +214,11 @@ private class HomeTileAdapter(
         }
     }
 
+    override fun getItemViewType(position: Int) = when (getItemAtPosition(position)) {
+        is BundledHomeTile -> BUNDLED_TILE_VIEW_TYPE
+        else -> CUSTOM_TILE_VIEW_TYPE
```

Comment:
You should explicitly use `is CustomHomeTile` and avoid using `else`. Else removes the purpose of making a list exhaustive because when you add new types, they'll all fall under the else case rather than forcing a compile error. Else also prevents you from considering edge cases you may not have thought about (e.g. if you're checking for 0 or 1, you could do `if (val > 0) thing(); else otherThing();` but you might forget the negative value case.